pkg/apiserver/handlers.go

```diff
@@ -17,21 +20,58 @@ func (s *Server) StartJob(ctx context.Context, req *pb.StartJobRequest) (*pb.Sta
 		// NOTE: Not supposed to happen as already checked, but check anyway.
 		return nil, fmt.Errorf("getUserFromContext: %w", err)
 	}
+	// Contextcheck // False positive. We don't want to use the request context to start the job in the background.
 	jobID, err := s.jobmanager.StartJob(user, req.GetCommand(), req.GetArgs())
 	if err != nil {
-		return nil, fmt.Errorf("jobmanager start job: : %w", err)
+		return nil, fmt.Errorf("job manager start job: : %w", err)
 	}
 	return &pb.StartJobResponse{JobId: jobID.String()}, nil
 }
 
-func (s *Server) StopJob(_ context.Context, _ *pb.StopJobRequest) (*pb.StopJobResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method StopJob not implemented")
+func (s *Server) StopJob(_ context.Context, req *pb.StopJobRequest) (*pb.StopJobResponse, error) {
+	jobID, _ := uuid.Parse(req.GetJobId()) // Already validated in middleware.
+
+	if err := s.jobmanager.StopJob(jobID); err != nil {
+		return nil, fmt.Errorf("job manager stop job: %w", err)
+	}
+
+	return &pb.StopJobResponse{}, nil
 }
 
-func (s *Server) GetJobStatus(_ context.Context, _ *pb.GetJobStatusRequest) (*pb.GetJobStatusResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method GetJobStatus not implemented")
+func (s *Server) GetJobStatus(_ context.Context, req *pb.GetJobStatusRequest) (*pb.GetJobStatusResponse, error) {
+	jobID, _ := uuid.Parse(req.GetJobId()) // Already validated in middleware.
+	job, err := s.jobmanager.LookupJob(jobID)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "lookup job: %s", err)
+	}
+	resp := &pb.GetJobStatusResponse{Status: job.Status()}
+	if resp.GetStatus() != pb.JobStatus_JOB_STATUS_RUNNING {
+		exitCode := int32(job.ExitCode()) //nolint:gosec // False positive.
+		resp.ExitCode = &exitCode
+	}
+	return resp, nil
 }
 
-func (s *Server) StreamLogs(_ *pb.StreamLogsRequest, _ grpc.ServerStreamingServer[pb.StreamLogsResponse]) error {
-	return status.Errorf(codes.Unimplemented, "method StreamLogs not implemented")
+func (s *Server) StreamLogs(req *pb.StreamLogsRequest, ss grpc.ServerStreamingServer[pb.StreamLogsResponse]) error {
+	ctx := ss.Context()
+	jobID, _ := uuid.Parse(req.GetJobId()) // Already validated in middleware.
+
+	r, err := s.jobmanager.StreamLogs(ctx, jobID)
+	if err != nil {
+		return fmt.Errorf("job manager stream logs: %w", err)
+	}
+
+	buf := make([]byte, 32*1024) //nolint:mnd // Default value from io.Copy, reasonable.
+loop:
```

Comment:
Use a `for {}`? Same for others.

Reply:
Why though? In the previous PR you requested to reduce indentation. Having a for{} here would just add a level of indentation without much benefit.

Reply:
A few reasons:

1. This is a block, it just doesn't like so visually
2. The label/goto idiom is rare in practice, which I expect would cause weirdness for most readers
3. This might encourage more sophisticated goto uses sneaking into the codebase, at which point we may get into real goto readability downsides

Note that this isn't a "goto bad", but I also don't see the appeal in replacing the more usual `for {}` construct.